qiskit_experiments/framework/composite/parallel_experiment.py

```diff
@@ -70,95 +72,111 @@ def __init__(
-                            # We skip delay instructions to allow for this.
-                            if inst.name == "delay":
-                                continue
-                            raise QiskitError(
-                                "Component experiment has been transpiled outside of the "
-                                "allowed physical qubits for that component. Check the "
-                                "experiment is valid on the backends coupling map."
-                            ) from ex
-                        mapped_cargs = [
-                            circuit.clbits[clbits[sub_circ.find_bit(i).index]] for i in cargs
+        sub_qubits = 0
+        for exp_idx, sub_exp in enumerate(self._experiments):
+            # Generate transpiled subcircuits
+            sub_circuits = sub_exp._transpiled_circuits()
+
+            # Qubit remapping for non-transpiled circuits
+            if not device_layout:
+                qubits = list(range(sub_qubits, sub_qubits + sub_exp.num_qubits))
+                qargs_map = {q: qubits[i] for i, q in enumerate(sub_exp.physical_qubits)}
+                sub_qubits += sub_exp.num_qubits
+            else:
+                qubits = list(sub_exp.physical_qubits)
+                qargs_map = {q: q for q in sub_exp.physical_qubits}
+
+            for circ_idx, sub_circ in enumerate(sub_circuits):
+                if circ_idx >= len(joint_circuits):
+                    # Initialize new joint circuit or extract
+                    # existing circuit if already initialized
+                    new_circuit = QuantumCircuit(num_qubits, name=f"parallel_exp_{circ_idx}")
+                    new_circuit.metadata = {
+                        "experiment_type": self._type,
+                        "composite_index": [],
+                        "composite_metadata": [],
+                        "composite_qubits": [],
+                        "composite_clbits": [],
+                    }
+                    joint_circuits.append(new_circuit)
+
+                # Add classical registers required by subcircuit
+                circuit = joint_circuits[circ_idx]
+                num_clbits = circuit.num_clbits
+                sub_clbits = sub_circ.num_clbits
+                clbits = list(range(num_clbits, num_clbits + sub_clbits))
+                if sub_clbits:
+                    creg = ClassicalRegister(sub_clbits)
+                    sub_cargs = [Clbit(creg, i) for i in range(sub_clbits)]
+                    circuit.add_register(creg)
+                else:
+                    sub_cargs = []
+
+                # Apply transpiled subcircuit
+                # Note that this assumes the circuit was not expanded to use
+                # any qubits outside the specified physical qubits
+                for inst, qargs, cargs in sub_circ.data:
+                    mapped_cargs = [sub_cargs[sub_circ.find_bit(i).index] for i in cargs]
```

Comment:
Does using the LRU cache further improve the performance? Intuitively we often repeatedly apply gates to the same wire.

Reply:
It would depend on the details of the `find_bit` method in terra, in principle it could be dict lookup which should have similar performance to lru_cache, but im not sure how they did it.

Reply:
I checked implementation and it is just a dict lookup. So generation of sub map should be enough fast.